modules/docs/arrow-docs/README.md

```diff
@@ -1,4 +1,110 @@
-# How to test
+# Arrow docs
+
+Arrow documentation deployed to the website.
+
+## How to contribute
+
+In case you need to add a new docs section, you should go like this:
+
+### 1. Add a menu entry to your docs
+
+Go to `modules/docs/arrow-docs/docs/_data/menu.yml` and add a menu entry in the proper section, as in:
+
+```
+- title: Applicative
+  url: /docs/typeclasses/applicative/
+```
+   
+Check [this PR](https://github.com/arrow-kt/arrow/pull/1134/files) for a real example.
+
+### 2. Add your docs file
+
+Add your docs as a Markdown file inside the corresponding directory. You have directories for all the sections 
+available in docs under `modules/docs/arrow-docs/docs/docs/`.
+
+Let's say you want to add docs for a Type class, for instance `Applicative`. You'd need to add a README like [this one](https://github.com/arrow-kt/arrow/blob/master/modules/docs/arrow-docs/docs/docs/arrow/typeclasses/applicative/README.md).
+
+You'll find all the sections available in the [docs side menu](https://arrow-kt.io/docs/). As you can see, there's:
+* **Quick start:** Basically how to start using Arrow, including links to libraries, posts, talks, and sample projects.
+* **API docs:** This is the public API reference.
+* **Patterns:** This section is much more like blogposts. Each one describes real world problems and how to solve them using Arrow. Pretty instructive.
+* **Data Types:** Docs for all the data types included in Arrow.
+* **Type Classes:** Docs for all the type classes included in Arrow.
+* **Effects:** Docs for all the effects module constructs, including type classes and data types.
+* **Optics:** Docs for the fancy optics DSL and all the related types.
+* **Arrow Query Language:** A human readable DSL used to build up arrow programs without the need for more specific knowledge. Usable over any query system.
+* **Generic:** We have some stuff generated at compile time under our `generic` module. Here you'll find the docs for it.
+* **Integration:** Docs for Arrow integration with third party libraries.
+* **Free:** Docs for Arrow Free Monads constructs.
+* **Recursion schemes:** Docs on how arrow achieves recursion schemes.
+* **Legal:** Credits and Licenses.
+
+### 3. Add a permalink to your file
+
+Add the proper permalink to your docs on top position of your README file, as in:
+
+```
+---
+layout: docs
+title: Applicative
+permalink: /docs/arrow/typeclasses/applicative/
+redirect_from:
+  - /docs/typeclasses/applicative/
+---
+``` 
+   
+Again, take a look at [this PR](https://github.com/arrow-kt/arrow/pull/1134/files) for a real example.
+
```

Comment:
I would add a section here that there needs to be a small description and a link from the datatype or typeclass intro pages back to the doc, so people can scan them easily in a single page.